op-e2e/actions/l2_batcher.go

```diff
@@ -220,26 +229,54 @@ func (s *L2Batcher) ActL2BatchSubmit(t Testing, txOpts ...func(tx *types.Dynamic
 	require.NoError(t, err, "need l1 pending header for gas price estimation")
 	gasFeeCap := new(big.Int).Add(gasTipCap, new(big.Int).Mul(pendingHeader.BaseFee, big.NewInt(2)))
 
-	rawTx := &types.DynamicFeeTx{
-		ChainID:   s.rollupCfg.L1ChainID,
-		Nonce:     nonce,
-		To:        &s.rollupCfg.BatchInboxAddress,
-		GasTipCap: gasTipCap,
-		GasFeeCap: gasFeeCap,
-		Data:      data.Bytes(),
-	}
-	for _, opt := range txOpts {
-		opt(rawTx)
+	var txData types.TxData
+	if s.l2BatcherCfg.DataAvailabilityType == batcherFlags.CalldataType {
+		rawTx := &types.DynamicFeeTx{
+			ChainID:   s.rollupCfg.L1ChainID,
+			Nonce:     nonce,
+			To:        &s.rollupCfg.BatchInboxAddress,
+			GasTipCap: gasTipCap,
+			GasFeeCap: gasFeeCap,
+			Data:      data.Bytes(),
+		}
+		for _, opt := range txOpts {
+			opt(rawTx)
+		}
+
+		gas, err := core.IntrinsicGas(rawTx.Data, nil, false, true, true, false)
+		require.NoError(t, err, "need to compute intrinsic gas")
+		rawTx.Gas = gas
+		txData = rawTx
+	} else if s.l2BatcherCfg.DataAvailabilityType == batcherFlags.BlobsType {
+		var b eth.Blob
+		require.NoError(t, b.FromData(data.Bytes()), "must turn data into blob")
+		sidecar, blobHashes, err := txmgr.MakeSidecar([]*eth.Blob{&b})
+		require.NoError(t, err)
+		require.NotNil(t, pendingHeader.ExcessBlobGas, "need L1 header with 4844 properties")
+		blobBaseFee := eip4844.CalcBlobFee(*pendingHeader.ExcessBlobGas)
+		txData = &types.BlobTx{
+			To:         s.rollupCfg.BatchInboxAddress,
+			Data:       nil,
+			Gas:        params.TxGas, // intrinsic gas only
+			BlobHashes: blobHashes,
+			Sidecar:    sidecar,
+			ChainID:    uint256.MustFromBig(s.rollupCfg.L1ChainID),
+			GasTipCap:  uint256.MustFromBig(gasTipCap),
+			GasFeeCap:  uint256.MustFromBig(gasFeeCap),
+			BlobFeeCap: new(uint256.Int).Mul(uint256.NewInt(2), uint256.MustFromBig(blobBaseFee)),
```

Comment:
I'm not too familiar with action tests, but if these transactions go to an actual geth mempool then you'll also want to make sure the blob fee cap is greater than the geth default min blob tx fee of 1 Gwei.